lib/Sabberworm/CSS/RuleSet/RuleSet.php

```diff
@@ -170,4 +199,21 @@ public function setComments(array $aComments) {
 		$this->aComments = $aComments;
 	}
 
+	/**
+	 * Returns all comments that were declared before this Rule
+	 * @return Comment[]
+	 */
+	public function getCommentsBefore()
```

Comment:
I’m not up-to-speed with the comment parsing routines but the way I believe this should work is that the comments associated with this RuleSet would always appear before. If a comment appeared inside, it would be associated with a `Rule`. And if it were to appear below the RuleSet, it would be parsed with the next RuleSet.

I’m not sure, though, how comments are associated that appear in an otherwise empty document or after the last RuleSet. IMHO, they should be stored on the `Document`, not simply discarded.

I guess my question is: why is this method needed, can’t we simply output all comments?